File: app/src/main/java/com/paradigmadigital/dagger/ui/main/MainActivity.java
package com.paradigmadigital.dagger.ui.main;

import com.paradigmadigital.dagger.R;
import com.paradigmadigital.dagger.ui.BaseActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import javax.inject.Inject;

import eu.sesma.library.LibraryMain;

public class MainActivity extends BaseActivity {

    @Inject
    MainPresenter presenter;

    View.OnClickListener listener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            presenter.onClick();
        }
    };

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getActivityComponent().inject(this);

        presenter.initialize();
        setTitle(presenter.getAppCollaboratorVersion());
        ((Button) findViewById(R.id.button)).setOnClickListener(listener);

        LibraryMain libraryMain = new LibraryMain();
        libraryMain.initialize();

        ((TextView)findViewById(R.id.textView2)).setText(presenter.getLibraryCollaboratorGreet());

        ((TextView)findViewById(R.id.textView3)).setText(presenter.getCoreCollaboratorGreet());
    }
}

File: module_main/src/main/java/eu/sesma/dagger/main/MainActivity.kt
package eu.sesma.dagger.main

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.TextView
import eu.sesma.dagger.R
import eu.sesma.dagger.core.di.getCoreComponent
import eu.sesma.dagger.main.di.DaggerMainActivityComponent
import eu.sesma.dagger.main.di.MainActivityModule
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var presenter: MainPresenter

    private val listener: View.OnClickListener = View.OnClickListener { presenter.onClick() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        DaggerMainActivityComponent.builder()
                .mainActivityModule(MainActivityModule(this))
                .coreComponent(getCoreComponent(applicationContext))
                .build()
                .inject(this)

        presenter.initialize()

        findViewById<View>(R.id.button).setOnClickListener(listener)

        findViewById<TextView>(R.id.textView1).text = presenter.coreCollaboratorGreet

        findViewById<TextView>(R.id.textView2).text = presenter.coreSingletonCollaboratorGreet

        findViewById<TextView>(R.id.textView3).text = presenter.mainActivityCollaboratorGreet

        findViewById<TextView>(R.id.textView4).text = presenter.mainActivityScopedCollaboratorGreet
    }
}
